### PR TITLE
Farmer bot disclaimer

### PR DESCRIPTION
**Pull request by SalmaElsoly**
### Description

Describe the changes introduced by this PR and what does it affect

### Changes

List of changes this PR includes
- add flag auto-approve to run sub command
- add disclaimer prompt to be called in run sub command

### Related Issues

List of related issues
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/1009#issue-2287864406

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: farmerbot/cmd/run.go [hunk 1]
```diff
@@ -55,3 +72,15 @@ var runCmd = &cobra.Command{
 		return farmerBot.Run(cmd.Context())
 	},
 }
+
+func disclaimerPrompt() (string, error) {
+	prompt := promptui.Select{
```
Comment (xmonader) [inline farmerbot/cmd/run.go]: I Love this!! however let's see if we can just get by using fmt.Print and fmt.Scanf ^_^

Comment (SalmaElsoly) [reply farmerbot/cmd/run.go]: okay

File: farmerbot/cmd/run.go [hunk 2]
```diff
@@ -15,7 +19,21 @@ var runCmd = &cobra.Command{
...
+		if err != nil {
+			return err
+		}
+		if !autoApprove {
+			answer, err := disclaimerPrompt()
+			if err != nil {
+				return err
+			}
+			if answer == "no" {
+				log.Info().Msg("Farmer bot is exiting ....")
```
Comment (rawdaGastan) [inline farmerbot/cmd/run.go]: make it `Farmerbot` to match other words

File: farmerbot/cmd/run.go [hunk 3]
```diff
@@ -55,3 +73,35 @@ var runCmd = &cobra.Command{
...
+	if _, err := fmt.Scanf("%s", &answer); err != nil {
+		return "", err
+	}
+
+	for i := 0; i < 3; i++ {
+		answer = strings.ToLower(answer)
+		if slices.Contains([]string{"yes", "no"}, answer) {
+			break
+		}
+		_, err := fmt.Print("\033[33mPlease enter yes or no only: \033[0m")
```
Comment (rawdaGastan) [inline farmerbot/cmd/run.go]: use `if _, err := ....`

**Review by rawdaGastan — CHANGES_REQUESTED** [review]
Good job 💯

File: farmerbot/cmd/run.go (hunk 2 above)
Comment (SalmaElsoly) [reply farmerbot/cmd/run.go]: done

File: farmerbot/cmd/run.go (hunk 3 above)
Comment (SalmaElsoly) [reply farmerbot/cmd/run.go]: done

**Comment by xmonader** [comment]
@Nabil-Salah Fix the CI please

**Comment by xmonader** [comment]
@Nabil-Salah i meant on a new branch from development branch (away from this ticket :) )

File: farmerbot/cmd/run.go [hunk 4]
```diff
@@ -55,3 +73,34 @@ var runCmd = &cobra.Command{
...
 	},
 }
+
+// disclaimerPrompt reads user input and return string, error
+func disclaimerPrompt() (string, error) {
+	var answer string
+	_, err := fmt.Print("\033[33mWarning: The Farmerbot is an optional feature developed by ThreeFold. Use at your own risk. While ThreeFold will do its best to fix any issues with the Farmerbot and minting, if minting is affected by the use of the Farmerbot, ThreeFold cannot be held responsible. Do you wish to proceed? (yes/no)\n\033[0m")
+	if err != nil {
+		return "", err
+	}
```
Comment (Eslam-Nawara) [inline farmerbot/cmd/run.go]:
```suggestion
	disclaimer := "\033[33mWarning: The Farmerbot is an optional feature developed by ThreeFold. Use at your own risk. While ThreeFold will do its best to fix any issues with the Farmerbot and minting, if minting is affected by the use of the Farmerbot, ThreeFold cannot be held responsible. Do you wish to proceed? (yes/no)\n\033[0m"
	if _, err := fmt.Print(disclaimer); err != nil {
		return "", err
	}
```

File: farmerbot/cmd/run.go [hunk 5]
```diff
@@ -55,3 +73,34 @@ var runCmd = &cobra.Command{
...
+		answer = strings.ToLower(answer)
+		if slices.Contains([]string{"yes", "no"}, answer) {
+			break
+		}
+		if _, err := fmt.Print("\033[33mPlease enter yes or no only: \033[0m"); err != nil {
+			return "", err
+		}
+		if _, err := fmt.Scanf("%s", &answer); err != nil {
+			return "", err
+		}
```
Comment (Eslam-Nawara) [inline farmerbot/cmd/run.go]: this results in asking the user for the answer 4 times, it's conventional to be only 3 times

File: farmerbot/cmd/run.go (hunk 4 above)
Comment (SalmaElsoly) [reply farmerbot/cmd/run.go]: done

File: farmerbot/cmd/run.go (hunk 5 above)
Comment (SalmaElsoly) [reply farmerbot/cmd/run.go]: I updated it to be only 3

File: farmerbot/cmd/run.go [hunk 6]
```diff
@@ -55,3 +73,35 @@ var runCmd = &cobra.Command{
...
+	}
+	answer = strings.ToLower(answer)
+	if !slices.Contains([]string{"yes", "no"}, answer) {
+		return "", fmt.Errorf("disclaimer expects yes or no, found : %s", answer)
+	}
```
Comment (Eslam-Nawara) [inline farmerbot/cmd/run.go]:
```suggestion
	for i := 0; i < 3; i++ {
		if _, err := fmt.Scanf("%s", &answer); err != nil {
			return "", err
		}

		answer = strings.ToLower(answer)
		if slices.Contains([]string{"yes", "no"}, answer) {
			break
		}
		if i < 2 {
			if _, err := fmt.Print("\033[33mPlease enter yes or no only: \033[0m"); err != nil {
				return "", err
			}
		}
	}

	if !slices.Contains([]string{"yes", "no"}, answer) {
		return "", fmt.Errorf("disclaimer expects yes or no, found : %s", answer)
	}

```

**Review by Eslam-Nawara — APPROVED** [review]

**Review by rawdaGastan — APPROVED** [review]

**Pull request merged**